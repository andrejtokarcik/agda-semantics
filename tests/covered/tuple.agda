module tuple where


data ℕ : Set where
  Z : ℕ 
  S : ℕ -> ℕ 


_+_ : ℕ -> ℕ ->  ℕ
n + Z   = n
n + S m = S (n + m)

infixr 30 _+_


data Nil : Set where
  [] : Nil


infix 20 _::_

data Cons (A B : Set) : Set where
  _::_ : A -> B -> Cons A B 


Tuple : Set -> ℕ -> Set
Tuple A Z     = Nil
Tuple A (S n) = Cons A (Tuple A n)


sumℕTuple : {n : ℕ} -> Tuple ℕ n -> Tuple ℕ n -> Tuple ℕ n
sumℕTuple {Z} []       []         = []
sumℕTuple {S n} (m :: l) (m' :: l') 
            = m + m' :: (sumℕTuple l l')




