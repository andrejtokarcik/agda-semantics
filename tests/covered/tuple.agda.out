<k>
    .K
</k>
<mgu>
    subst(.KList)
</mgu>
<ctx-stack>
    .List
</ctx-stack>
<sig>
    Fun ( "Tuple" ) |-> ( #symVariable(67) : Set ( 0 ) ) -> ( #symVariable
      (68) : Data ( "8469" ) ) -> Set ( 0 )
    Fun ( "_+_" ) |-> ( #symVariable(10) : Data ( "8469" ) ) -> ( #symVariable
      (11) : Data ( "8469" ) ) -> Data ( "8469" )
    Fun ( "sum8469Tuple" ) |-> { #symVariable(140) : Data ( "8469" ) } -> ( 
      #symVariable(138) : ((Fun ( "Tuple" ) Data ( "8469" )) #symVariable(140)
      ) ) -> ( #symVariable(139) : ((Fun ( "Tuple" ) Data ( "8469" )) 
      #symVariable(140)) ) -> ((Fun ( "Tuple" ) Data ( "8469" )) #symVariable(
      140))
    Data ( "8469" ) |-> Set ( 0 )
    Data ( "Cons" ) |-> ( #symVariable(54) : Set ( 0 ) ) -> ( #symVariable(53)
       : Set ( 0 ) ) -> Set ( 0 )
    Data ( "Nil" ) |-> Set ( 0 )
    Con ( "S" ) |-> ( #symVariable(9) : Data ( "8469" ) ) -> Data ( "8469" )
    Con ( "Z" ) |-> Data ( "8469" )
    Con ( "[]" ) |-> Data ( "Nil" )
    Con ( "_::_" ) |-> { #symVariable(66) : Set ( 0 ) } -> { #symVariable(65) 
      : Set ( 0 ) } -> ( #symVariable(63) : #symVariable(66) ) -> ( 
      #symVariable(64) : #symVariable(65) ) -> ((Data ( "Cons" ) #symVariable(
      66)) #symVariable(65))
</sig>
<ctx>
    .Map
</ctx>
<meta>
    .Map
</meta>
<ncat>
    Name ( "8469" ) |-> Data ( "8469" )
    Name ( "Cons" ) |-> Data ( "Cons" )
    Name ( "Nil" ) |-> Data ( "Nil" )
    Name ( "S" ) |-> Con ( "S" )
    Name ( "Tuple" ) |-> Fun ( "Tuple" )
    Name ( "Z" ) |-> Con ( "Z" )
    Name ( "[]" ) |-> Con ( "[]" )
    Name ( "_+_" ) |-> Fun ( "_+_" )
    Name ( "_::_" ) |-> Con ( "_::_" )
    Name ( "sum8469Tuple" ) |-> Fun ( "sum8469Tuple" )
</ncat>
<fun>
    Fun ( "Tuple" ) |-> 'fc(((#symFunClauses(80) Var ( "A" )) Con ( "Z" ))
       = Data ( "Nil" ),, ((#symFunClauses(113) Var ( "A" )) (Con ( "S" ) Var 
      ( "n" ))) = ((Data ( "Cons" ) Var ( "A" )) ((Fun ( "Tuple" ) Var ( "A" )
      ) Var ( "n" ))))
    Fun ( "_+_" ) |-> 'fc(((#symFunClauses(23) Var ( "n" )) Con ( "Z" )) = Var
       ( "n" ),, ((#symFunClauses(52) Var ( "n" )) (Con ( "S" ) Var ( "m" ))) 
      = (Con ( "S" ) ((Fun ( "_+_" ) Var ( "n" )) Var ( "m" ))))
    Fun ( "sum8469Tuple" ) |-> 'fc((((#symFunClauses(728) { Con ( "Z" ) }) Con
       ( "[]" )) Con ( "[]" )) = Con ( "[]" ),, (((#symFunClauses(2515) { (Con
       ( "S" ) Var ( "n" )) }) ((((Con ( "_::_" ) { Data ( "8469" ) }) { ((
      FunClauses ( Fun ( "Tuple" ) , 'fc(((#symFunClauses(927) #symVariable(
      928)) Con ( "Z" )) = Data ( "Nil" ),, ((#symFunClauses(929) #symVariable
      (930)) (Con ( "S" ) #symVariable(931))) = ((Data ( "Cons" ) #symVariable
      (930)) ((Fun ( "Tuple" ) #symVariable(930)) #symVariable(931)))) ) Data 
      ( "8469" )) Var ( "n" )) }) Var ( "m" )) Var ( "l" ))) ((((Con ( "_::_" 
      ) { Data ( "8469" ) }) { ((FunClauses ( Fun ( "Tuple" ) , 'fc(((
      #symFunClauses(1229) #symVariable(1230)) Con ( "Z" )) = Data ( "Nil" )
      ,, ((#symFunClauses(1231) #symVariable(1232)) (Con ( "S" ) #symVariable(
      1233))) = ((Data ( "Cons" ) #symVariable(1232)) ((Fun ( "Tuple" ) 
      #symVariable(1232)) #symVariable(1233)))) ) Data ( "8469" )) Var ( "n" )
      ) }) Var ( "m'" )) Var ( "l'" ))) = ((((Con ( "_::_" ) { Data ( "8469" )
       }) { ((Fun ( "Tuple" ) Data ( "8469" )) Var ( "n" )) }) ((Fun ( "_+_" )
       Var ( "m" )) Var ( "m'" ))) (((Fun ( "sum8469Tuple" ) { Var ( "n" ) }) 
      Var ( "l" )) Var ( "l'" ))))
</fun>
